### PR TITLE
tests: posix: Fix pthread attribute initialization

### LINKED ISSUE #7569: test: posix/pthread_rwlock
tests/posix/pthread_rwlock/portability.posix calls `pthread_attr_destroy()` on the uninitialized, stack allocated, `pthread_attr_t attr[N_THR];` leading to undefined behaviour:
```
starting test - test_rw_lock

main acquire WR lock and 3 threads acquire RD lock
==18028== Thread 5:
==18028== Conditional jump or move depends on uninitialised value(s)
==18028==    at 0x80507F5: zap_pthread_attr_destroy (pthread.c:533)
==18028==    by 0x8049D4E: test_rw_lock (posix_rwlock.c:83)
==18028==    by 0x80559FC: run_test_functions (ztest.c:56)
==18028==    by 0x8055BB7: test_cb (ztest.c:196)
==18028==  Uninitialised value was created by a stack allocation
==18028==    at 0x804982F: test_rw_lock (posix_rwlock.c:55)
```

### PR DESCRIPTION
**Pull request by ramakrishnapallala**
Fix pthread_rwlock test pthread attributes initialization.

Fixes #7569

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

### CODE REVIEW

File: tests/posix/pthread_rwlock/src/posix_rwlock.c [hunk 1]
```diff
@@ -80,7 +80,7 @@ static void test_rw_lock(void)
 
 	/* Creating N premptive threads in increasing order of priority */
 	for (i = 0; i < N_THR; i++) {
-		pthread_attr_destroy(&attr[i]);
+		memset(&attr[i], 0x0, sizeof(pthread_attr_t));
```
Comment (aescolar-ot) [inline tests/posix/pthread_rwlock/src/posix_rwlock.c]: No need for the memset, pthread_attr_init is the one meant to initialize the attr[i]

**Review by aescolar — CHANGES_REQUESTED** [review]

File: tests/posix/pthread_rwlock/src/posix_rwlock.c (hunk 1 above)
Comment (ramakrishnapallala) [reply tests/posix/pthread_rwlock/src/posix_rwlock.c]: this is just to make sure, if we are not initializing any parameters in the attr_init() this set the fields to 0x0.

Comment (aescolar-ot) [reply tests/posix/pthread_rwlock/src/posix_rwlock.c]: I understand that. I meant it is not, and should not be, necessary.

Comment (ramakrishnapallala) [reply tests/posix/pthread_rwlock/src/posix_rwlock.c]: Ok, will fix it.

**Comment by ramakrishnapallala** [comment]
PR updated.

**Review by aescolar — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**